scripts/remove-example-foundation.sh

```diff
@@ -0,0 +1,385 @@
+#!/usr/bin/env bash
+
+# Copyright 2022 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+################################################################################
+# remove-example-foundation.sh
+# Version: 0.1
+#
+# Remove all the resourses created by terraform-example-foundation deployment
+# with Google Cloud SDK.
+#
+# This tool is designed to work with default configuration only. if you deployed
+# terraform-example-foundation with custom configurations, this tool give you
+# unexpacted result. I am not responsible for any damages of any kind arising
+# from the use of this tool. Please use this tool at your own risk.
+
+# Resource deleting order
+# 1. Delete Firewall associations.
+# 2. Delete Firewall polices.
+# 3. Delete Billing Account budgets.
+# 4. Unlink Billing Account links.
+# 5. Remove principals from Billing Account.
+# 6. Delete Tags.
+# 7. Delete Liens.
+# 8. Delete Projets.
+# 9. Delete Folders.
+# 10. Revert org polices to default.
+# 11. Delete Groups
+# 12. Remove principals from organization.
+
+# gcloud version
+# Google Cloud SDK 436.0.0
+# alpha 2023.06.16
+# beta 2023.06.16
+# bq 2.0.93
+# core 2023.06.16
+# gcloud-crc32c 1.0.0
+# gsutil 5.24
+# terraform-tools 0.10.0
+
+################################################################################
+
+# 0. Initiate variables
+
+# Quota project ID
+QPJT_ID="REPLACEME"
+# QPJT_ID=$(gcloud config list --format="value(core.project)")
+
+# Target organization number
+ORG_NUM="REPLACEME"
+# ORG_NUM=$(gcloud organizations list --format="value(name)")
+
+# Billing account id
+BA_ID="REPLACEME"
+# BA_ID=$(gcloud beta billing accounts list --format="value(name)")
+
+# Configuration flags
+RESET_ORG_POLICES=false
+
+# Folder list from the organization
+FLDR_LIST=$(gcloud resource-manager folders list \
+    --organization "$ORG_NUM" --filter="displayName:fldr*" \
+    --format="value(name)")
+
+# Project list from the organization
+PJT_LIST=$(gcloud projects list --filter="projectId:prj-*" \
+    --format="value(projectId)")
+
+################################################################################
+
```

Comment:
I've found this script very useful for tearing down the example foundation :rocket:
 
I would like to contribute this block which checks the currently authenticated user for the Owner role on the organization. This might be too broad of a IAM permission but we could suggest removing it after running this script (or make the check more specific on the granualr IAM permissions): 

```shell
echo ""
echo "# 0. IAM check"

if echo "$(gcloud alpha organizations get-iam-policy $ORG_NUM )" | grep -q 'roles/owner'; then
    echo "You are properly configured with IAM."
else
    echo -e "Looks like you do not have the Owner role on organization($ORG_NUM)."
    exit 1
fi
```